core/gmail_accounts.py

```diff
@@ -0,0 +1,95 @@
+from httplib2 import Http
+
+from django.conf import settings
+from django.utils.crypto import get_random_string
+from apiclient.errors import HttpError
+from apiclient.discovery import build
+from oauth2client.service_account import ServiceAccountCredentials
+
+from core.models import Event
+
+
+GAPPS_JSON_CREDENTIALS = {
+  "type": "service_account",
+  "project_id": "djangogirls-website",
+  "private_key_id": settings.GAPPS_PRIVATE_KEY_ID,
+  "private_key": settings.GAPPS_PRIVATE_KEY,
+  "client_email": "django-girls-website@djangogirls-website.iam.gserviceaccount.com",
+  "client_id": "114585708723701029855",
+  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+  "token_uri": "https://accounts.google.com/o/oauth2/token",
+  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/django-girls-website%40djangogirls-website.iam.gserviceaccount.com"
+}
+
+
+def get_gapps_client():
+    credentials = ServiceAccountCredentials.from_json_keyfile_dict(
+        GAPPS_JSON_CREDENTIALS,
+        scopes=settings.GAPPS_ADMIN_SDK_SCOPES
+    )
+
+    delegated_credentials = credentials.create_delegated('hello@djangogirls.org')
+    http_auth = delegated_credentials.authorize(Http())
+
+    return build('admin', 'directory_v1', http=http_auth)
+
+
+def make_email(slug):
+    """Get the email address for the given slug"""
+    return '%s@djangogirls.org' % slug
+
+
+def create_gmail_account(event):
+    """
+    Create a new account
+    """
+    email = event.email
+    password = get_random_string(length=10)
```

Comment:
Will they be required to change the password after first login? If not maybe we could use the password generator from Django :)